andela/ah-scorpion
# #159053991 Allow Users to Favourite and  Unfavourite Article

#### What does this PR do
- Allows users to favourite and unfavourite an article

#### Description of tasks done
- added DB models for favouriting article
- Added urls endpoint for favouriting article
- added views to handle favouriting and un favouriting
- made migrations for database model changes
- added unit tests for this functionality

#### How can this PR be manually tested
Prerequisite: Postman
- use git pull to get the source code from ft-favourite-article-159053991 branch
- use `source .env` to export environmental variables and activate virtual env
- run `python manage.py migrate` to apply recent migrations
- run python manage.py runserver` to start  the project
- if not already done, create a new user account and login
- if not already done, create an article and copy the `slug`
- create a `POST` request to the following endpoint `<str:slug>/favorite/` replacing the `<slug>` with the slug you copied above
- you should get the article back. If you try again you should get a message `You have already marked this article as a favourite`
- create a `DELETE` request to the following endpoint `<str:slug>/favorite/` replacing the `<slug>` with the slug you copied above
- you should get the article back. If you try again you should get a message `You have not marked this article as a favourite`
- if you use a slug that is not valid, you should get a message `The article was not found` on both requests

#### Related PT stories
- [#159053991](https://www.pivotaltracker.com/story/show/159053991)
